modules/encoders/php/base64.rb

```diff
@@ -20,12 +20,21 @@ def initialize
   end
 
   def encode_block(state, buf)
-    # Have to have these for the decoder stub, so if they're not available,
-    # there's nothing we can do here.
     %w[c h r ( ) . e v a l b a s e 6 4 _ d e c o d e ;].uniq.each do |c|
       raise BadcharError if state.badchars.include?(c)
     end
 
+    executor = "eval"
+    %w[e v a l].uniq.each do |c|
+      if state.badchars.include?(c)
+        %w[a s s e r t].uniq.each do |c|
```

Comment:
Checking for the characters `a`, `e`, and `r` is a bit redundant here given we already check those on line 25/23, so there is no need to check them again. Only the characters `s` and `t` are unknown if they are in the bad character list or not by this point, so over half of this loop is repeated work.